### PR TITLE
Fix ORC read corruption when specified schema does not match file order

### PR DESCRIPTION
**Pull request by wbo4958**
This PR first fixes the orc data reading mess up for the schema which can't be pruned
second, adds the dis-order read schema unit tests for both Parquet and Orc.

Please refer to issue #3060. 

With this PR, GPU can produce the correct result but The CPU itself has produced the wrong result. So this PR will bring the in-compatible behavior between GPU and CPU.

### CODE REVIEW

**Comment by wbo4958** [comment]
build

**Comment by wbo4958** [comment]
build

File: tests/src/test/scala/com/nvidia/spark/rapids/OrcScanSuite.scala [hunk 1]
```diff
@@ -33,4 +34,66 @@ class OrcScanSuite extends SparkQueryCompareTestSuite {
...
+      StructField("c1_int", IntegerType))))) { frame => frame }
+
+  testSparkResultsAreEqual("schema-can-prune dis-order read schema 5",
+    frameFromOrcWithSchema("schema-can-prune.orc", StructType(Seq(
+      StructField("c3_long", LongType),
+      StructField("c1_int", IntegerType))))) { frame => frame }
+
+  /**
+   * We can't compare the results from CPU and GPU, since CPU will get in-correct result
+   * see https://github.com/NVIDIA/spark-rapids/issues/3060
```
Comment (jlowe) [inline tests/src/test/scala/com/nvidia/spark/rapids/OrcScanSuite.scala:73]: IIUC the "can't be pruned" case is when the column names appear to be the same as the case when ORC doesn't provide column names (which was true for old versions of ORC in the early Hive days).  In those cases, I don't think Spark can fix this because there aren't any column names to match up if the specified schema doesn't match the file order.

**Review by jlowe — APPROVED** [review]
@wbo4958 can you elaborate in the PR description about the binary files that were added?

File: tests/src/test/scala/com/nvidia/spark/rapids/OrcScanSuite.scala (hunk 1 above)
Comment (wbo4958) [reply tests/src/test/scala/com/nvidia/spark/rapids/OrcScanSuite.scala:73]: Yes, you're correct. So if it's not a Spark bug, Should I merge this PR? @jlowe

Comment (firestarman) [reply tests/src/test/scala/com/nvidia/spark/rapids/OrcScanSuite.scala:73]: Per your test case, even without this fix, the CPU and GPU still produce different outputs. So personally I prefer to merge it.

Comment (jlowe) [reply tests/src/test/scala/com/nvidia/spark/rapids/OrcScanSuite.scala:73]: Agreed, this seems to not break anything and improves behavior in a corner case.
